newIDE/app/src/MainFrame/index.js

```diff
@@ -304,6 +312,22 @@ class MainFrame extends React.Component<Props, State> {
 
   openFromPathOrURL = (url: string, cb: Function) => {
     const { i18n } = this.props;
+
+    const autoSavePath = url + '.autosave';
+    if (fs.existsSync(autoSavePath)) {
```

Comment:
So the logic seems ok to me, but we should move this to LocalProjectOpener. The idea is that here you're refering to some capabilities like `fs` which are only available when running on Electron (this is probably crashing the webapp when trying to open a project, because fs is null).
Almost everything that is specific to Electron is abstracted in files called `LocalSomethingSomething.js`.

By moving all the read/write of files to LocalProjectOpener.js and LocalProjectWriter.js, we allow ourselves to later create a new abstraction, for example DropboxProjectOpener and DropboxProjectWriter (or Google Drive, etc... you see the idea 😉).
It also reduce the logic written in MainFrame/index.js - which is already a huge component and should be kept free of anything related to "low level" things like writing/reading files or dealing with autosaves :)

Reply:
funny thing - I initially had it in LocalProjectOpener, but for whatever reason moved it to main xD I will move it back